atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/TransactionsLockWatchEvents.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.api.watch;
+
+import java.util.List;
+import java.util.Map;
+
+import org.immutables.value.Value;
+
+import com.palantir.lock.watch.LockWatchEvent;
+import com.palantir.lock.watch.LockWatchStateUpdate;
+
+/**
+ * Represents a condensed view of lock watch events occurring between some known version and a set of start transaction
+ * calls.
+ */
+public interface TransactionsLockWatchEvents {
+    <T> T accept(Visitor<T> visitor);
+
+    interface Visitor<T> {
+        T visit(Success success);
+        T visit(Failure failure);
+    }
```

Comment:
These names are a bit close to the `LockWatchStateUpdate` ones, and a little confusingly there a `Snapshot` becomes here a `Failure`.  Maybe `Events` rather than `Success`, and something like `Snapshot` or `State` rather than `Failure`?